erts/emulator/drivers/common/efile_drv.c

```diff
@@ -1261,26 +1261,61 @@ static void free_read_line(void *data)
     EF_FREE(d);
 }
 
+void read_file_zero_size(struct t_data* d);
+
+/* [ERL-327] Some special files like /proc/... have reported size 0 */
+void read_file_zero_size(struct t_data* d) {
+    size_t actual_size = 0;
+    while (1) {
+        ssize_t read_result;
+        char* datap = 0;
```

Comment:
It should be possible to use `d->c.read_file.orig_bytes` and `d->c.read_file.offset` instead of `actual_size` and `datap` to reduce the number of variables and to make the code more similar to the non-0-size code (less places where the type could get wrong)